libvmi/driver/xen/xen.c

```diff
@@ -2455,15 +2455,51 @@ xen_set_vcpureg(
 #if defined(ARM32) || defined(ARM64)
     return xen_set_vcpureg_arm(vmi, value, reg, vcpu);
 #elif defined(I386) || defined (X86_64)
-    if (!xen_get_instance(vmi)->hvm) {
-        if (8 == xen_get_instance(vmi)->addr_width) {
-            return xen_set_vcpureg_pv64(vmi, value, reg, vcpu);
-        } else {
-            return xen_set_vcpureg_pv32(vmi, value, reg, vcpu);
+    if (xen_get_instance(vmi)->hvm) {
+        switch (reg) {
+        case RAX:
+        case RBX:
+        case RCX:
+        case RDX:
+        case RBP:
+        case RSI:
+        case RDI:
+        case RSP:
+        case R8:
+        case R9:
+        case R10:
+        case R11:
+        case R12:
+        case R13:
+        case R14:
+        case R15:
+        case RIP:
+        case RFLAGS:
+        case CR0:
+        case CR2:
+        case CR3:
```

Comment:
So it seems `xen_pfn_to_cr3_x86_64(value >> 12)` just left-shifts it back 12 bits, essentially zeroing out the bottom 12 bits. I ran the following code about 1000 times with no issues, so unless there is some need for the bottom 12 bits of CR3, it shouldn't affect anything

```
reg_t tmp;
vmi_get_vcpureg(vmi, &tmp, CR3, event->vcpu_id);
vmi_set_vcpureg(vmi, tmp, CR3, event->vcpu_id);
```